Remove 'track-remove-by-setting-innerHTML.html' local copy in favor of WPT test case

Ahmad-S792 opened the pull request:
#### 402cf346e35431f919808ad30ef892288d090a5a
<pre>
Remove &apos;track-remove-by-setting-innerHTML.html&apos; local copy in favor of WPT test case

<a href="https://bugs.webkit.org/show_bug.cgi?id=120470">https://bugs.webkit.org/show_bug.cgi?id=120470</a>

Reviewed by Eric Carlson.

This patch removes local copy of W3C test in favor of Web Platform Test.
I confirmed locally via running &apos;import-w3c-tests&apos; that we have latest copy of test in tree.
Similarly, I removed test expectation since WPT indicates it as passing test as of STP 181.

* LayoutTests/media/track/track-remove-by-setting-innerHTML.html: Deleted
* LayoutTests/media/track/track-remove-by-setting-innerHTML-expected.txt: Ditto
* LayoutTests/TestExpectations: Unskip &apos;WPT&apos; test
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/track/track-element/track-remove-by-setting-innerHTML-expected.txt: Added Test Expectation
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-element/track/track-element/track-remove-by-setting-innerHTML-expected.txt: Add Platform Specific Expectation

Canonical link: <a href="https://commits.webkit.org/270001@main">https://commits.webkit.org/270001@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ae94c7a5ebae9ac5b8cb05335867b724d54627a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24061 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2173 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25149 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26201 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22166 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/3796 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/24544 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22644 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24305 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/1704 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/20799 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/26790 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1461 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/21717 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/27963 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/21936 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22004 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/25739 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1396 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/19071 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/1417 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5807 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/1806 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/1749 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/track/track-element/track-remove-by-setting-innerHTML.html:
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!DOCTYPE html><!-- webkit-test-runner [ captionDisplayMode=automatic ] -->
```
Please do not use `<!-- webkit-test-runner` directives in WPT

Ahmad-S792 replied:
It is needed for webvtt related tests. :-( Although in this case, it is not, so I will not remove it.

Ahmad-S792 committed:
Remove 'track-remove-by-setting-innerHTML.html' local copy in favor of WPT test case

https://bugs.webkit.org/show_bug.cgi?id=120470

Reviewed by Eric Carlson.

This patch removes local copy of W3C test in favor of Web Platform Test.
I confirmed locally via running 'import-w3c-tests' that we have latest copy of test in tree.
Similarly, I removed test expectation since WPT indicates it as passing test as of STP 181.

* LayoutTests/media/track/track-remove-by-setting-innerHTML.html: Deleted
* LayoutTests/media/track/track-remove-by-setting-innerHTML-expected.txt: Ditto
* LayoutTests/TestExpectations: Unskip 'WPT' test
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-elements/track/track-element/track-remove-by-setting-innerHTML-expected.txt: Added Test Expectation
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/semantics/embedded-content/media-element/track/track-element/track-remove-by-setting-innerHTML-expected.txt: Add Platform Specific Expectation

Canonical link: https://commits.webkit.org/270001@main
